jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/GlobalTimerServiceVolumeTest.java

```diff
@@ -0,0 +1,268 @@
+import org.kie.internal.task.api.TaskModelProvider;
+import org.kie.internal.task.api.UserGroupCallback;
+import org.kie.internal.task.api.model.InternalOrganizationalEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GlobalTimerServiceVolumeTest extends TimerBaseTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(GlobalTimerServiceVolumeTest.class);
+    private UserGroupCallback userGroupCallback;
+
+    private GlobalSchedulerService globalScheduler;
+
+    private RuntimeManager manager;
+
+    private EntityManagerFactory emf;
+
+    @Before
+    public void setup() {
+
+        Properties properties = new Properties();
+        properties.setProperty("mary", "HR");
+        properties.setProperty("john", "HR");
+        userGroupCallback = new JBossUserGroupCallbackImpl(properties);
+
+        System.setProperty("org.quartz.properties", "quartz-db.properties");
+        testCreateQuartzSchema();
+        globalScheduler = new QuartzSchedulerService();
+
+        emf = Persistence.createEntityManagerFactory("org.jbpm.test.persistence");
+    }
+
+    @After
+    public void tearDown() {
+        globalScheduler.shutdown();
+        if (manager != null) {
+            manager.close();
+        }
+        emf.close();
+    }
+
+    @Test(timeout=30000)
+    public void testPerProcessInstance() throws Exception {
+        testStrategy(1);
+    }
+    
+    @Test(timeout=30000)
+    public void testPerRequest() throws Exception {
+        testStrategy(2);
+    }
+    
+    @Test(timeout=30000)
+    public void testPerCase() throws Exception {
+        testStrategy(3);
+    }
+    
+    private void testStrategy(int strategy) throws Exception {
+        int numberOfProcesses = 10;
+        CountDownProcessEventListener countDownListener = new CountDownProcessEventListener("timer", numberOfProcesses);
+        // prepare listener to assert results
+        final List<Long> timerExporations = new ArrayList<Long>();
+        ProcessEventListener listener = new DefaultProcessEventListener(){
+
+            @Override
+            public void afterNodeLeft(ProcessNodeLeftEvent event) {
+                if (event.getNodeInstance().getNodeName().equals("timer")) {
+                    timerExporations.add(event.getProcessInstance().getId());
+                }
+            }
+            
+        };
+        RuntimeEnvironment environment = RuntimeEnvironmentBuilder.Factory.get()
+                .newDefaultBuilder()
+                .entityManagerFactory(emf)
+                .userGroupCallback(userGroupCallback)
+                .addAsset(ResourceFactory.newClassPathResource("org/jbpm/test/functional/timer/IntermediateCatchEventTimerCycleWithHT3.bpmn2"), ResourceType.BPMN2)
+                .schedulerService(globalScheduler)
+                .registerableItemsFactory(new TestRegisterableItemsFactory(listener, countDownListener))
+                .get();
+
+        if (strategy == 1) {
```

Comment:
I think this test could be designed much cleaner as a parameterized test, putting the code above into a `@Before` method and making the `manager` instance variable - the test parameter.